src/services/EdFi.DataManagementService.Api/ApiSchema/ProjectSchema.cs

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Api.ApiSchema.Extensions;
+using EdFi.DataManagementService.Api.ApiSchema.Model;
+
+namespace EdFi.DataManagementService.Api.ApiSchema;
+
+/// <summary>
+/// Provides information from the ProjectSchema portion of an ApiSchema.json document
+/// </summary>
+public class ProjectSchema(JsonNode _projectSchemaNode)
+{
+    private readonly Lazy<MetaEdProjectName> _projectName = new(() =>
+    {
+        return new MetaEdProjectName(_projectSchemaNode.SelectRequiredNodeFromPathAs<string>("$.projectName"));
+    });
+
+    /// <summary>
+    /// The ProjectName for this ProjectSchema, taken from the projectName
+    /// </summary>
+    public MetaEdProjectName ProjectName => _projectName.Value;
```

Comment:
This seems like the way to memoize in C#

Reply:
Yes, exposing a private instance property as a public property...